src/app.rs

```diff
@@ -37,6 +39,24 @@ impl ToResponseCode for InclusionProofResponse {
     }
 }
 
+#[derive(Serialize)]
+pub enum VerifySemaphoreProofResponse {
```

Comment:
Sorry to be that guy but can we talk a bit about the format of this response? I find that serde's default enum serialization is kind of ugly, the `Ok` response serializes to:

```rust
{
    "Ok": {
        "root": "0x0000000000000000000000000000000000000000000000000000000000000000",
        "status": "pending",
        "pending_valid_as_of": "2023-03-07T19:51:01Z",
        "mined_valid_as_of": null
    }
}
```

whereas any of the error responses serializes to a string.

If we go with something like
```rust
#[derive(Serialize)]
#[serde(rename_all = "camelCase")]
pub enum VerifySemaphoreProofResponse {
    Ok(RootItem),
    Err(VerifySemaphoreProofResponseError),
}

#[derive(Serialize)]
#[serde(rename_all = "camelCase")]
pub enum VerifySemaphoreProofResponseError {
    InvalidRoot,
    InvalidProof,
    ProverError,
}
```

then it'll serialize to either
```json
{
    "ok": {
        "root": "0x0000000000000000000000000000000000000000000000000000000000000000",
        "status": "pending",
        "pending_valid_as_of": "2023-03-07T19:51:01Z",
        "mined_valid_as_of": null
    }
}
```

or

```json
{
    "err": "invalidRoot"
}
```

Reply:
I made `VerifySemaphoreProofResponse` transparent and moved errors to the parent result object.